### ISSUE TITLE
Filter JAWS and NV Access support

### ISSUE THREAD

**Issue by rcelis** [gregnb/mui-datatables#1730]
Filter headers ARIA compliance

## Expected Behavior
When displaying the filters popup JAWS or NV Access should be able to read the labels for each filter option

## Current Behavior
JAWS and NV Access are unable to read the labels for each filtering option, they just read the content of the field

## Steps to Reproduce (for bugs)

1. Open a page that has a MUI-Datatable with at least a column that you can filter by
2. Enable JAWS or NV Access
3. Click the toolbar filter icon
4. Hear that JAWS or NV Access are unable to read the label of the filter option(s)

## Your Environment

| Tech         | Version |
|--------------|---------|
| Material-UI  |  4.11.1  |
| MUI-datatables  |  3.7.7  |
| React        |  17.0.5  |
| Browser      |  Firefox, Chrome, Chromium Edge   |
| JAWS          | 2021.2105.53 |
| NV Access  |   2020.4   |
